### DISCUSSION TITLE
peer for nested elements

### DISCUSSION THREAD

**Discussion by devcustrom** [tailwindlabs/tailwindcss#5996] [Ideas]
I see it this way
https://play.tailwindcss.com/aryzWob4Rp

**Comment by marckohlbrugge** [comment]
@devcustrom Thanks for sharing your code! I was able to solve a related problem: https://play.tailwindcss.com/K4X5I0I37T

Curious to hear if you found a better workaround since? For example using the `addVariant` functionality

**Reply by marckohlbrugge** [reply]
I was able to do it with `addVariant` 🎉

```js
// tailwind.config.js
module.exports = {
  theme: {
    extend: {
      // ...
    },
  },
  plugins: [
    function ({ addVariant }) {
      addVariant('peer-group-valid', '.peer:valid ~ .peer-group &')
    },
  ],
}
```

Demo: https://play.tailwindcss.com/oCy6ORP8yW?file=config

I'm pretty sure you can do the same for other modifiers like `disabled`

**Reply by devcustrom** [reply]
With the release of version 3.1, this is no longer a problem for me, since custom selectors have appeared there
